nsdb-common/src/main/scala/io/radicalbit/nsdb/common/statement/SQLStatement.scala

```diff
@@ -147,6 +147,25 @@ case class DescOrderOperator(override val dimension: String) extends OrderOperat
   */
 case class LimitOperator(value: Int)
 
+/**
+  * Timestamp to wrap timestamp value for tracking relative timestamp
+  */
+sealed trait TimestampValue {
+  def timestamp: Long
+}
+// consider if have to change all the timestamp (now are Long) to AbsoluteTimestampValue
+// case class AbsoluteTimestampValue(override val timestamp: Long) extends TimestampValue
```

Comment:
yes. We must think about using this class, and also about making this more generic

Reply:
I was thinking that, instead of having an ADT related to the timestamp, we can have a relative ADT for comparisons (equality included) with a generic type.
e.g. 

```
sealed trait ComparisonValue[T] {
  def value: [T]
}

case class RelativeComparisonValue[T](override val value: [T], operator: String, quantity: [T], unitMeasure: String)`
case class AbsoluteTimestampValue[T](override val value: Long)
```

Then this classes can be updated as follows

```
case class ComparisonExpression[T](dimension: String, comparison: ComparisonOperator, value: ComparisonValue[T]) extends Expression
case class EqualityExpression[T](dimension: String, value: ComparisonValue[T]) extends Expression
```

This is just a proposal, any suggestion is welcome